react/src/components/BookList/index.js

```diff
@@ -0,0 +1,36 @@
+import React, { Component } from 'react';
+import BookItem from './components/BookItem';
+import {data} from '../../mock.js';
+
+class BookList extends Component {
+  render() {
+    const filterText = this.props.filterText;
+    const filterField = this.props.filterField;
+    let books = data.map(function(book, thisArg){
+      if(
+        !filterText ||
+        !filterField ||
+        (filterField === "1" && book.title.toLowerCase().includes(filterText.toLowerCase())) ||
+        (filterField === "2" && book.author.toLowerCase().includes(filterText.toLowerCase()))
```

Comment:
Both `1` and `2` should belong to a constant

Reply:
Using that `filterField` can contain identifiers that helps us to distinct which filterType is active, we may do something like:
```
Rename 1 -> 'title'
Rename 2 -> 'author'
```
```
if (
  !filterText ||
  !filterField ||
  book[filterField].toLowerCase().includes(filterText.toLowerCase())
) {
  ...
}
```
What do you think ?